enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,302 @@
+
+In order to keep these metrics in sync with those of OpenShift we must have
+the following:
+* CI jobs deploying MicroShift must use Telemetry.
+* CI jobs Telemetry must the same endpoints as production MicroShift
+  deployments.
+* CI jobs Telemetry data must be visible in a Grafana dashboard.
+* E2E tests must be included to check for both sending and reception of
+  Telemetry data.
+
+### Sending metrics
+OpenShift is sending metrics through this API every 4.5 minutes. For MicroShift
+deployments this might be a bit excessive due to resource usage, network
+traffic and usage patterns.
+
+MicroShift should send data once a day to minimize network traffic as it can be
+deployed in constrained environments. To allow customizations a new
+configuration option shall be added. This option will drive how often a metrics
+payload should be sent and it will default to 24h. In every MicroShift start
+all metrics shall be sent. Afterwards, it will follow the configuration option
+to schedule the next send.
+
+As described above MicroShift will be using the [direct request](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#metricsv1receive-endpoint-receive-metrics-in-prompbwriterequest-format-from-any-client)
+endpoint.
+Each metric must follow [Prometheus WriteRequest](https://github.com/prometheus/prometheus/blob/release-2.38/prompb/remote.proto#L22)
+format.
+
+### Sampling and batching metrics
+Metrics sent by MicroShift can be categorized in:
+* Static metrics. These are fixed throughout the execution of MicroShift, such
+as memory or CPU.
+* Dynamic metrics. These change and evolve throughout the execution of
+MicroShift, such as resource usage or resources in the cluster.
+
+If we assume default values, metrics are sent at least once a day in those
+deployments that do not disable the functionality. While this is good enough
+for static metrics, it provides a degraded view for dynamic ones. Having
+once-per-day data on resource usage makes it virtually impossible to extract
+any patterns out of it. For this reason it might be beneficial to sample
+dynamic metrics more often and then batch them together with the static
+metrics when sending them.
+
+Sampling metrics means MicroShift needs to store values in between metrics
+reports. The dynamic metrics are not numerous enough to take a toll on resource
+usage, but this interval must be configurable and default to a sensible value.
+
+As seen before each metric is under 250B. Considering 7 metrics, having one of
+them hold multiple samples (one for each resource type, totalling ) a complete
+metrics payload with static and dynamic metrics would be under 2 KB.
+Using defaults (1 payload every 24h, with 1h samples for all dynamic metrics)
+would be under 40 KB.
+In order to not lose sampling data MicroShift should not rely on storing it
+only in memory. The `/var/lib/microshift` directory should be used to store
+temporary data about usage for sending it later.
```

Comment:
I'm not sure to understand correctly but if the intention is to buffer samples locally and send them in a batch every 24h, it won't work against the actual Telemetry requirements: it can't ingest samples "too far" in the past (I think that the backfill window is 2h).

Backfilling data into Telemetry would be interesting but it isn't something that is supported right now.

Reply:
This has been fixed by reporting every hour and remove any sampling/batching. Now MicroShift should be able to comply to Telemetry requirements.